core/drivers/cbmem_console.c

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2023, Google Limited
+ */
+
+#include <compiler.h>
+#include <console.h>
+#include <drivers/cbmem_console.h>
+#include <drivers/serial.h>
+#include <io.h>
+#include <keep.h>
+#include <kernel/dt.h>
+#include <libfdt.h>
+#include <mm/core_mmu.h>
+#include <types_ext.h>
+#include <util.h>
+
+#define CURSOR_MASK (BIT(28) - 1)
+#define OVERFLOW BIT(31)
+
+struct cbmem_console {
+	uint32_t size;
+	uint32_t cursor;
+	uint8_t body[0];
+} __packed;
```

Comment:
Why is this `__packed`?

Reply:
I got that from here: https://github.com/coreboot/coreboot/blob/master/payloads/libpayload/drivers/cbmem_console.c

But clearly I can't justify keeping it because I'm not using it in the other structures that have similar 32-bit value alignment requirements. I'll remove it.